src/components/cesium_map/cesiumh3.js

```diff
@@ -53,13 +55,20 @@ export class H3Grid {
         }
         this.rect_str = rstr;
         this.loading = true;
+        // read q and fq solr query param values
+        let qArray = solrQuery.getQFQSolrQueryParamValues(store.getState()['query']).q.split(",")
+        let fqArray = solrQuery.getQFQSolrQueryParamValues(store.getState()['query']).fq.split(",")
         let url = new URL(this._service + "/");
         if (this.rect_str === GLOBAL_RECT1 || this.rect_str === GLOBAL_RECT2) { 
             url.searchParams.set('resolution', 1); // globe view resolution value should be small
         } else {
             url.searchParams.set('bb', this.rect_str);
             url.searchParams.set('resolution', 5);
         }
+        // convert it to q query param values
+        let queryArray = [...qArray, ...fqArray];
+        let combinedQuery = queryArray.join(' AND ');
```

Comment:
Concatenate the q and fq query params into a single q query param for filtering.